### PR TITLE
Ionq connection

### PR DESCRIPTION
**Pull request by ValentinS4t1qbit**
IonQ API python wrapper class and tests, to facilitate the submission of experiments through their portal.
Comes with a notebook.

### CODE REVIEW

File: tangelo/linq/tests/test_ionq_connection.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+"""
+    A test class to check that features related to the Honeywell API are behaving as expected.
```
Comment (AlexandreF-1qbit) [inline tangelo/linq/tests/test_ionq_connection.py:2]: IonQ

File: examples/linq/2.qpu_connection.ipynb [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+   "metadata": {},
+   "source": [
+    "# Using quantum cloud services API with Tangelo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook elaborates on how the `tangelo.linq` module can facilitate hardware experiments by integrating the API provided by some hardware providers (IonQ, Honeywell...) or broader quantum cloud services providers such as Azure Quantum or Braket. We essentially provide convenience wrappers to these APIss: the API of your favorite cloud provider is still called underneath, using your own credentials.\n",
```
Comment (AlexandreF-1qbit) [inline examples/linq/2.qpu_connection.ipynb:14]: APIss -> APIs?

File: examples/linq/2.qpu_connection.ipynb [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. IonQ Rest API <a class=\"anchor\" id=\"1\"></a>\n",
+    "\n",
+    "\n",
+    "### 1.1 References <a class=\"anchor\" id=\"1.1\"></a>\n",
+    "\n",
+    "Tangelo provides a few Python wrappers around the REST API provided by IonQ, to facilitate job submission and monitoring. The most up-to-date list of supported features and information regarding the IonQ API is available at https://docs.ionq.co\n",
```
Comment (AlexandreF-1qbit) [inline examples/linq/2.qpu_connection.ipynb:54]: `.co` -> `.com`

File: examples/linq/2.qpu_connection.ipynb [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"IONQ_APIKEY\"] = '2T14z1YQEzMLCwuYM110oXPDT2h850E4'"
```
Comment (AlexandreF-1qbit) [inline examples/linq/2.qpu_connection.ipynb:73]: Is it a valid key? Should we leave it there?

File: examples/linq/2.qpu_connection.ipynb [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+    "ionq_connection = IonQConnection()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Backend info\n",
+    "\n",
+    "Tangelo's `get_backend_info` method allows us to return some information about all the devices listed on IonQ's platform. Tangelo currently returns this info inside a `pandas.Dataframe` object, to help with visualization. "
```
Comment (AlexandreF-1qbit) [inline examples/linq/2.qpu_connection.ipynb:150]: Dataframe -> DataFrame

**Review by AlexandreF-1qbit — CHANGES_REQUESTED** [review]

File: examples/linq/2.qpu_connection.ipynb (hunk 4 above)
Comment (ValentinS4t1qbit) [reply examples/linq/2.qpu_connection.ipynb:73]: It's not a valid key :)

File: tangelo/linq/tests/test_ionq_connection.py (hunk 1 above)
Comment (ValentinS4t1qbit) [reply tangelo/linq/tests/test_ionq_connection.py]: lol.

File: examples/linq/2.qpu_connection.ipynb (hunk 3 above)
Comment (ValentinS4t1qbit) [reply examples/linq/2.qpu_connection.ipynb:54]: No, it is .co

**Review by AlexandreF-1qbit — APPROVED** [review]

**Review by JamesB-1qbit — APPROVED** [review]

**Pull request merged**